.travis.yml

```diff
@@ -1,10 +1,36 @@
-language: java
-before_install:
-  - bash dev/travis-before_install.sh
+language: generic
+
+services:
+  - docker
 
 env:
-  - HADOOP_PROFILE=default TEST_CODECS=uncompressed,brotli
-  - HADOOP_PROFILE=default TEST_CODECS=gzip,snappy
+  global:
+    - JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
+    - IMAGE=parquet-mr
+    - CACHE_FOLDER=$HOME/docker-images
+    - CACHE_FILE=${CACHE_FOLDER}/parquet-mr.tgz
+  matrix:
+    - TEST_CODECS=uncompressed,brotli
+    - TEST_CODECS=gzip,snappy
+
+cache:
+  directories:
+    - "$HOME/.m2"
+    - ${CACHE_FOLDER}
+
+before_install:
+  - if [[ -f ${CACHE_FILE} ]]; then
+      docker load -i "${CACHE_FILE}";
+    else
+      docker build -f dev/Dockerfile -t "${IMAGE}" "${PWD}";
+      mkdir -p "${CACHE_FOLDER}";
+      docker save "${IMAGE}" | gzip -c > "${CACHE_FILE}";
+    fi
+
+install: docker run -v $HOME/.m2:/root/.m2 -v ${PWD}:/parquet-mr -e JAVA_HOME=$JAVA_HOME -e TEST_CODECS=$TEST_CODECS -i -t parquet-mr /bin/bash -c "cd /parquet-mr && mvn install --batch-mode -DskipTests=true -Dmaven.javadoc.skip=true -Dsource.skip=true | pv -fbi 60 > mvn_install.log || (cat mvn_install.log && false)"
```

Comment:
I would make the command part of the container as well. For example, use the `WORKDIR` for setting the path.

Why do we `-DskipTests=true`?

Reply:
Same comment that above, it is better to have a more flexible container than to tight commands into it for our use case.

Reply:
Notice that htis is the install step and the goal here is only to put the artifcats into the .m2 directory that's the reason I found, but this code is inherited from the previous build file, as well as the one in the script step.

Reply:
I would definitely prefer to make both a single line but I wanted to stay equal until this was working and then discuss about how to improve it, e.g. the way we call the IT with an environment variable can be done as a maven failsafe execution step too, to only have one run and cover everything.

Reply:
Makes sense, thanks